File: app/src/main/java/se/gustavkarlsson/skylight/android/dagger/modules/replaceable/BackgroundUpdateTimeoutModule.java
package se.gustavkarlsson.skylight.android.dagger.modules.replaceable;

import org.threeten.bp.Duration;

import javax.inject.Named;

import dagger.Module;
import dagger.Provides;
import dagger.Reusable;

import static se.gustavkarlsson.skylight.android.dagger.Names.BACKGROUND_UPDATE_TIMEOUT_NAME;

@Module
public abstract class BackgroundUpdateTimeoutModule {

	// Published
	@Provides
	@Reusable
	@Named(BACKGROUND_UPDATE_TIMEOUT_NAME)
	static Duration provideBackgroundUpdateTimeout() {
		return Duration.ofSeconds(30);
	}

}

File: app/src/main/kotlin/se/gustavkarlsson/skylight/android/dagger/modules/replaceable/BackgroundUpdateTimeoutModule.kt
package se.gustavkarlsson.skylight.android.dagger.modules.replaceable

import dagger.Module
import dagger.Provides
import dagger.Reusable
import org.threeten.bp.Duration
import se.gustavkarlsson.skylight.android.dagger.BACKGROUND_UPDATE_TIMEOUT_NAME
import javax.inject.Named

@Module
class BackgroundUpdateTimeoutModule {

    // Published
    @Provides
    @Reusable
    @Named(BACKGROUND_UPDATE_TIMEOUT_NAME)
    fun provideBackgroundUpdateTimeout(): Duration {
        return Duration.ofSeconds(30)
    }

}
